### PR TITLE
BZ-1310439 - RuntimeManagerFactory may fail to initialize in OSGi

### PR DESCRIPTION
**Pull request by mswiderski**
depends on https://github.com/droolsjbpm/droolsjbpm-knowledge/pull/133

### CODE REVIEW

File: jbpm-flow-builder/src/main/java/org/jbpm/osgi/flow/compiler/Activator.java [hunk 1]
```diff
@@ -35,7 +36,8 @@ public void start(BundleContext bc) throws Exception {
         this.processBuilderReg = bc.registerService( new String[]{ ProcessBuilderFactoryService.class.getName(), Service.class.getName()},
                                                                    new ProcessBuilderFactoryServiceImpl(),
                                                                    new Hashtable() );
-        ProcessBuilderFactory.reInitializeProvider();
+        RuntimeManagerFactory.Factory.reset();
+        ProcessBuilderFactory.reInitializeProvider();
```
Comment (psiroky) [inline jbpm-flow-builder/src/main/java/org/jbpm/osgi/flow/compiler/Activator.java]: Could we use the same method name, co be consistent? The `reInitializeProvider()` may not be ideal, but we could change that as well potentially.

Comment (mswiderski) [inline jbpm-flow-builder/src/main/java/org/jbpm/osgi/flow/compiler/Activator.java]: not really as they do different things. RuntimeManagerFactory ... reset() does only reset of the flag and exception without reinitializing the factory impl itself

Comment (psiroky) [inline jbpm-flow-builder/src/main/java/org/jbpm/osgi/flow/compiler/Activator.java]: Ah, thanks. I see now. However, that brings the next question -- should we unify the approach? It seems that the the end result should be the same (a way to re-try to load the provider class)

Comment (mswiderski) [inline jbpm-flow-builder/src/main/java/org/jbpm/osgi/flow/compiler/Activator.java]: we could and actually that was my first attempt though when jbpm-flow or jbpm-flow-builder is started there might still not be all stuff available - in this case jbpm-runtime-manager so it just makes sure it resets and allows "clean start" for clients calls. While the other makes sense to eagerly init as it's possible.

Comment (psiroky) [inline jbpm-flow-builder/src/main/java/org/jbpm/osgi/flow/compiler/Activator.java]: Just thinking out loud -- would it be better to add the call to `jbpm-runtime-manager`'s Activator then?

Comment (mswiderski) [inline jbpm-flow-builder/src/main/java/org/jbpm/osgi/flow/compiler/Activator.java]: there is no such Activator class in jbpm-runtime-manager...

Comment (psiroky) [inline jbpm-flow-builder/src/main/java/org/jbpm/osgi/flow/compiler/Activator.java]: Right, so I think we should create one. That would make it consist with the other jBPM modules. I can take a look at that.

**Pull request closed without merging**